components/service/pocket/src/main/java/mozilla/components/service/pocket/api/PocketJSONParser.kt

```diff
@@ -54,6 +56,30 @@ internal class PocketJSONParser {
         null
     }
 
+    /**
+     * Validate and return the Pocket story imageUrl only if it has an actual image path.
+     * The JSON object representing a Pocket story might contains a url representing the `imageUrl`
+     * but that might not be valid.
+     *
+     * @param [jsonStory] JSON respecting the schema of a Pocket story. Expected to contain [JSON_STORY_IMAGE_URL_KEY].
+     *
+     * @return the full imageUrl from [jsonStory] if it does not end in "/null".
+     *
+     * @throws JSONException if a value for [JSON_STORY_IMAGE_URL_KEY] is not found or ends with "/null".
+     */
+    @Throws(JSONException::class)
+    @VisibleForTesting
+    internal fun getValidImageUrl(jsonStory: JSONObject): String {
+        val imageUrl = jsonStory.getString(JSON_STORY_IMAGE_URL_KEY)
+        val imagePath = imageUrl.substringAfterLast("/")
+
+        if (imagePath == "null") {
+            throw JSONException("Null image path")
+        }
```

Comment:
The solution from this patch is hardcoded for a specific scenario - if the imageUrl ends with "/null".
(we were already dropping stories with the entire imageUrl being `null`).
Will try to get someone from Pocket to review on whether this is enough or there might be other scenarios we'd need to account for.